### PR TITLE
Add sphinx documentation for v4 scripts and update doc strings accordingly

### PR DESCRIPTION
**Pull request by jkgoodrich**
All changes are just to work with creating the documentation correctly, and some small formatting changes.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -39,3 +39,59 @@ jobs:
...
       - name: Check docstrings
         run: pydocstyle --match-dir='(?!v2|cuKING).*' gnomad_qc
+  docs:
+    name: Build documentation
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
```
Comment (ch-kr) [inline .github/workflows/ci.yml:49]: what does this line do?

File: docs/index.rst [hunk 2]
```diff
@@ -0,0 +1,41 @@
+gnomad_qc
+=========
+This is the documentation for the complete set of scripts used to perform sample and
+variant QC for the gnomAD v4 release. We are continuously updating and improving upon
+the code to handle new releases as they grow in size and complexity and as they require
+increasingly sophisticated QC treatment. The current code therefore represents the most
+recent iteration of our pipelines and is guaranteed to change over time.
+
+The scripts make reference to gnomAD-related metadata files (not public) and may
```
Comment (ch-kr) [inline docs/index.rst:9]:
```suggestion
The scripts reference gnomAD-related metadata files (not public) and may
```
optional (slightly clearer)

File: docs/index.rst [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+A more extensive overview and explanation of the gnomAD QC process is available on the
+`gnomAD browser <https://gnomad.broadinstitute.org/news/>`_ and may help inform users’
+design decisions for other pipelines.
+
+  * `v2.1 <https://gnomad.broadinstitute.org/news/2018-10-gnomad-v2-1/>`_
+  * `v3 <https://gnomad.broadinstitute.org/news/2019-10-gnomad-v3-0/>`_
+  * `v3.1 <https://gnomad.broadinstitute.org/news/2020-10-gnomad-v3-1-new-content-methods-annotations-and-data-availability/>`_
+  * `v4.0 <https://gnomad.broadinstitute.org/news/2023-11-gnomad-v4-0/>`_
+
+Note also that many basic functions and file paths used in the code are imported from a
```
Comment (ch-kr) [inline docs/index.rst:30]: should this note be higher up in the docs?

File: docs/index.rst (hunk 2 above)
Comment (ch-kr) [reply docs/index.rst:9]: also maybe along these lines: should we mention that our QC is incredibly tailored to Broad-sequenced/processed data?

File: gnomad_qc/v4/annotations/vrs_annotation_batch.py [hunk 4]
```diff
@@ -1,24 +1,27 @@
...
---header-path gs://gnomad/v4.0/annotations/exomes/vrs-header-fix.txt \
---run-vrs \
---annotate-original \
---overwrite \
---backend-mode batch \
---data-type exomes \
---test
+Example command to use:
+
+    .. code-block:: bash
```
Comment (ch-kr) [inline gnomad_qc/v4/annotations/vrs_annotation_batch.py:9]: ooh, cool!

File: gnomad_qc/v4/variant_qc/vqsr.py [hunk 5]
```diff
@@ -1358,14 +1359,14 @@ def main(args):
...
-        default=VQSR_FEATURES["snv"],
+        default=VQSR_FEATURES["exomes"]["snv"],
         type=str,
         nargs="+",
     )
     parser.add_argument(
         "--indel-features",
         help="Features to use in the indel VQSR model.",
-        default=VQSR_FEATURES["indel"],
+        default=VQSR_FEATURES["exomes"]["indel"],
```
Comment (ch-kr) [inline gnomad_qc/v4/variant_qc/vqsr.py:1369]: was the exome/genome stratification added after this script was run?

File: docs/README.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
+# Documentation
+
+Documentation for gnomad is generated using [Sphinx](https://www.sphinx-doc.org/en/master/).
```
Comment (ch-kr) [inline docs/README.md:3]: should we update gnomad to gnomad_qc (just to differentiate between this repo and gnomad_methods)?

**Review by ch-kr — COMMENTED** [review]
thank you for adding! I have some small questions and suggetions.

I also just wanted to note that I didn't review all of the formatting changes very closely 😅

File: .github/workflows/ci.yml (hunk 1 above)
Comment (jkgoodrich) [reply .github/workflows/ci.yml:49]: allows us to import from `gnomad_qc.sample_qc.cuking`

File: docs/README.md (hunk 6 above)
Comment (jkgoodrich) [reply docs/README.md:3]: oops, yup, thank you!

File: docs/index.rst (hunk 2 above)
Comment (jkgoodrich) [reply docs/index.rst:9]: yeah we can, to make this fast, it was really mostly a copy and paste of the repo readme

File: docs/index.rst (hunk 3 above)
Comment (jkgoodrich) [reply docs/index.rst:30]: sure

File: gnomad_qc/v4/variant_qc/vqsr.py (hunk 5 above)
Comment (jkgoodrich) [reply gnomad_qc/v4/variant_qc/vqsr.py:1369]: yup

**Review by ch-kr — APPROVED** [review]
LGTM!

**Pull request merged**